serf.py

```diff
@@ -73,51 +73,89 @@ def lumberjack(self):
             pass
 
     def no_profession(self):
-        if self.target_coords:
+        if self.target_coords is not None:
             changes = utilities.get_vector(self, self.target_coords[0], self.target_coords[1], self.rect.x + 15, self.rect.y + 22)
             self.change_x = changes[0]
             self.change_y = changes[1]
-        if not self.collide_check():
-            self.move(self.current_map)
+            if not self.collide_check():
+                self.move(self.current_map)
 
     def collide_check(self):
+        # hypothetical copy of ourself
         projection = pygame.sprite.Sprite()
         projection.image = self.image
-        projection.rect = self.image.get_rect()
-        projection.rect.x = self.rect.x
-        projection.rect.y = self.rect.y
-        projection.rect.x += self.change_x
-        projection.rect.y += self.change_y
-
-        future_collisions = None
-        future_collisions = (pygame.sprite.spritecollide(self, self.current_map.units, False))
-        future_collisions.remove(self)
+        projection.rect = deepcopy(self.rect)
+
+        # hypothetically move where we want to go
+        projection.rect.x += self.change_x * 2
+        projection.rect.y += self.change_y * 2 
```

Comment:
This solves the bug but I don't *really* know why. This leaves a 2px gap between units.